docs/source/example_notebooks/dowhy_functional_api.ipynb

```diff
@@ -182,10 +188,30 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "# We will still need CausalModel as the Functional Effect Estimation is still Work-In-Progress\n",
-    "causal_model = CausalModel(data=data[\"df\"], treatment=treatment_name, outcome=outcome_name, graph=data[\"gml_graph\"])\n",
+    "# Basic Estimate Effect function\n",
+    "\n",
     "\n",
-    "estimate = causal_model.estimate_effect(identified_estimand, method_name=\"backdoor.propensity_score_matching\")\n",
+    "propensity_score_estimator = PropensityScoreMatchingEstimator(\n",
+    "    data=data[\"df\"],\n",
+    "    identified_estimand=identified_estimand,\n",
+    "    treatment=treatment_name,\n",
+    "    outcome=outcome_name,\n",
+    "    control_value=0,\n",
+    "    treatment_value=1,\n",
+    "    test_significance=None,\n",
+    "    evaluate_effect_strength=False,\n",
+    "    confidence_intervals=False,\n",
+    "    target_units=\"ate\",\n",
+    "    effect_modifiers=graph.get_effect_modifiers(treatment_name, outcome_name),\n",
+    ")\n",
+    "\n",
+    "estimate = estimate_effect(\n",
+    "    treatment=treatment_name,\n",
+    "    outcome=outcome_name,\n",
+    "    identified_estimand=identified_estimand,\n",
+    "    identifier_name=\"backdoor\",\n",
+    "    method=propensity_score_estimator,\n",
+    ")\n",
```

Comment:
Or maybe I'm misunderstanding this, and the whole point of the function `estimate_effect` is to take care of calling the _methods_ `fit` and `estimate_effect`. Then never mind my comment above.

Reply:
Hey @petergtz, I believe that what you just described is my end goal here, I think that @amit-sharma still wants to keep the `estimate_effect` function as a way to automatically select parameters (for users that maybe don't know which parameters to pick) I want to separate this into several PRs to avoid creating one giant PR and also keep testing for backwards compatibility. This first PR is just extracting the `estimate_effect` function to make it "functional". My next PR will be introducing the `fit()` method and moving parameters around to the place they are actually used (and removing the some of the `**kwargs`).

Reply:
That's great! Thanks for providing the context, @andresmor-ms. Resolving.